app/helpers/validation_helper.php

```diff
@@ -0,0 +1,21 @@
+<?php
+function validate_between($str, $min, $max)
+{
+    $length = mb_strlen($str); // count characters based on encoding
+    return $length <= $max && $length >= $min;
+}
+
+function validate_username($str)
+{
+    return preg_match('/^[[:alnum:]_]*$/', $str);
+}
+
+function validate_unique_name($username)
+{
+    return !User::getByUsername($username);
```

Comment:
If I just catch the exception, the username's uniqueness will not be evaluated with the other fields. So if the other fields did not pass the evaluation, the username's uniqueness will not be evaluated and an error will not be shown even if it already exists po.

![screen shot 2015-09-02 at 13 52 45](https://cloud.githubusercontent.com/assets/13476640/9623922/2dae169c-517a-11e5-9432-cac86416e8fc.png)